### ISSUE TITLE
Get rid of `classproperty`

### ISSUE THREAD

**Issue by pipermerriam** [ethereum/trinity#1057]
### What is wrong?

I think the `classproperty` is problematic and should be removed.

It results in us having `Any` types for the return values and while I'm sure there are ways to fix this I think we're fixing the wrong problem.

### How can it be fixed

Replace all of the usage of `classproperty` with `classmethod` versions of them as getter functions.

```python
class Thing:
    @classproperty
    def some_thing(cls):
        ...
```

This becomes:

```python
class Thing:
    @classmethod
    def get_some_thing(cls):
        ...
```

**Comment by pipermerriam** [comment]
There's an alternate approach which could be *ok*.  If extracted into it's own package, it could also implement the `mypy` plugin API to preserve type safety on the original `classmethod`.

**Comment by veox** [comment]
I thonk this should've been closed by PR #1099.
